pkg/web/src/app/queries/fetchHelpers.ts

```diff
@@ -1,100 +1,112 @@
-import { INetworkContext, useNetworkContext } from '@app/common/context/NetworkContext';
-import { QueryFunction } from 'react-query/types/core/types';
-import { useHistory } from 'react-router-dom';
-import { History, LocationState } from 'history';
-import { useClientInstance } from '@app/client/helpers';
 import { KubeResource } from '@migtools/lib-ui';
 import { IKubeResponse, IKubeStatus } from '@app/client/types';
-import { ENV } from '@app/common/constants';
+import { consoleFetch } from '@openshift-console/dynamic-plugin-sdk';
+import { META } from '@app/common/constants';
 
-interface IFetchContext {
-  history: History<LocationState>;
-  checkExpiry: INetworkContext['checkExpiry'];
-  currentUser: INetworkContext['currentUser'];
-}
+/** Simulate an axios fetch call */
+const authorizedK8sRequest = async <T>({
+  method = 'GET',
+  url,
+  data,
+  opetions = {},
+}: {
+  method?: 'GET' | 'POST' | 'PATCH' | 'PUT' | 'DELETE';
+  url: string;
+  data?: object;
+  opetions?: object;
+}): Promise<IKubeResponse<T>> => {
+  const headers =
+    method === 'PATCH' && data
+      ? { 'Content-Type': 'application/merge-patch+json', Accept: 'application/json' }
+      : { 'Content-Type': 'application/json', Accept: 'application/json' };
+  const body = method !== 'GET' && data ? JSON.stringify(data) : undefined;
 
-export const useFetchContext = (): IFetchContext => {
-  const { checkExpiry, currentUser } = useNetworkContext();
-  return { history: useHistory(), checkExpiry, currentUser };
-};
-
-export const authorizedFetch = async <TResponse, TData = unknown>(
-  url: string,
-  fetchContext: IFetchContext,
-  extraHeaders: RequestInit['headers'] = {},
-  method: 'get' | 'post' = 'get',
-  returnMode: 'json' | 'blob' = 'json',
-  bypassRedirect = false,
-  data?: TData
-): Promise<TResponse> => {
-  const { history, checkExpiry } = fetchContext;
+  const requestOptions = {
+    ...opetions,
+    method: method,
+    headers: headers,
+    body: body,
+  };
 
-  if (ENV.AUTH_REQUIRED !== 'false' && fetchContext.currentUser?.access_token) {
-    extraHeaders['Authorization'] = `Bearer ${fetchContext.currentUser.access_token}`;
-  }
+  const response = await consoleFetch(url, requestOptions);
```

Comment:
why we are not using `consoleFetchJSON()` ?

Reply:
`consoleFetchJSON` returns the parsed data, and filter out the status parts, `consoleFetch` returns the `Response`.
When mocking axios or vanila `fetch` it's easier to start from a `Response`, if we where using `consoleFetchJSON` we will need to re-create the status.